File: output/StarCoder/codenet/Python/Java/s269975433.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int M = scanner.nextInt();
        int Q = scanner.nextInt();

        List<int[]> qGroups = new ArrayList<>();
        for (int i = 0; i < Q; i++) {
            int[] group = {scanner.nextInt(), scanner.nextInt(), scanner.nextInt(), scanner.nextInt()};
            qGroups.add(group);
        }

        int maxScore = 0;
        for (int[] p : CombinationsCombinations.combinationsWithReplacement(M, N)) {
            int tempScore = 0;
            for (int[] q : qGroups) {
                if (p[q[1] - 1] - p[q[0] - 1] == q[2]) {
                    tempScore += q[3];
                }
            }
            maxScore = Math.max(maxScore, tempScore);
        }

        System.out.println(maxScore);
    }
}

class CombinationsCombinations {
    public static List<int[]> combinationsWithReplacement(int n, int k) {
        List<int[]> result = new ArrayList<>();
        combinationWithReplacementHelper(result, new int[k], n, k, 0, 0);
        return result;
    }

    private static void combinationWithReplacementHelper(List<int[]> result, int[] data, int n, int k, int start, int index) {
        if (index == k) {
            result.add(data.clone());
            return;
        }
        for (int i = start; i <= n - k + index + 1; i++) {
            data[index] = i + 1;
            combinationWithReplacementHelper(result, data, n, k, i, index + 1);
        }
    }
}

File: dataset_reduzido/codenet/Python/Code/s269975433.py
from itertools import combinations_with_replacement
import sys

def LI(): return map(int, sys.stdin.readline().rstrip().split())

N, M, Q = LI()

Q_groups = []
for _ in range(Q):
    Q_groups.append([*LI()])

max_score = 0
for p in combinations_with_replacement(range(1, M + 1), r=N):
    temp_score = 0
    for q in Q_groups:
        if p[q[1] - 1] - p[q[0] - 1] == q[2]:
            temp_score += q[3]
    max_score = max(max_score, temp_score)

print(max_score)